UPGRADE-4.0.md

```diff
@@ -0,0 +1,57 @@
+# Upgrading from 3.x to 4.0
+
+### 1. Update dependency in composer.json
+In require-dev section change the version constraint:
+
+```diff
+-        "lmc/coding-standard": "^3.3",
++        "lmc/coding-standard": "^4.0",
+```
+
+Then run `composer update`.
+
+### 2. Configuration updates
+
+Configuration now uses ECSConfig class instead of ContainerConfigurator. Update your `ecs.php` to use the new configuration style:
+
+```diff
+-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+-
+-return static function (ContainerConfigurator $containerConfigurator): void {
++use Symplify\EasyCodingStandard\Config\ECSConfig;
++
++return ECSConfig::configure()
+```
+
+Rules are now set using `ECSConfig::configure()->withRules([])` or `ECSConfig::configure()->withConfiguredRule()` instead of `$services->set()`.
+Skiping tests is now done using `ECSConfig::configure()->withSkip()` instead of `$parameters->set(Option::SKIP, ...)`.
+Imports are now done using `ECSConfig::configure()->withSets()` instead of `$containerConfigurator->import()`.
+
+See [ECS documentation](https://github.com/easy-coding-standard/easy-coding-standard/tree/main?tab=readme-ov-file#configure) for more configuration options and examples.
```

Comment:
There is a reference to `withConfiguredRule` in the sentences before, but where is this mentioned in the ECS documentation? I was not able to find anything about it...

Reply:
Yeah, it looks like it's not documented and I don't remember how I found it, probably via IDE when I typed `->`  and saw what IDE suggests :D